common/src/contract-params.test.ts

```diff
@@ -0,0 +1,474 @@
+}))
+
+jest.mock('common/chart', () => ({
+  ...jest.requireActual('common/chart'),
+  serializeMultiPoints: jest.fn().mockReturnValue({
+    'answer-123': [[1, 0.5]],
+  }),
+}))
+
+jest.mock('common/supabase/contracts', () => ({
+  ...jest.requireActual('common/supabase/contracts'),
+  getContractPageViews: jest.fn().mockResolvedValue(100),
+}))
+
+jest.mock('common/supabase/chart-annotations', () => ({
+  getChartAnnotations: jest.fn().mockResolvedValue([
+    {
+      id: 'annotation-1',
+      contract_id: '123',
+      event_time: '2022-01-01T00:00:00Z',
+      content: 'Test annotation',
+    },
+  ]),
+}))
+
+describe('getContractParams', () => {
+  afterEach(() => {
+    // Restore all mocks to their original implementations after each test
+    jest.restoreAllMocks()
+  })
+  it('should return not authed state for unauthorized access', async () => {
+    const mockDb = createClient('instanceId', 'key')
+    const mockContract = {
+      id: '123',
+      mechanism: 'cpmm-1',
+      outcomeType: 'BINARY',
+      visibility: 'private',
+      deleted: false,
+    } as Contract
+
+    const result = await getContractParams(
+      mockContract,
+      mockDb,
+      true,
+      undefined
+    )
+
+    expect(result).toEqual({
+      state: 'not authed',
+      slug: mockContract.slug,
+      visibility: mockContract.visibility,
+    })
+  })
+
+  it('should return not found state for a deleted contract with unauthorized access', async () => {
+    const mockDb = createClient('instanceId', 'key')
+    const mockContract = {
+      id: '123',
+      mechanism: 'cpmm-1',
+      outcomeType: 'BINARY',
+      visibility: 'private',
+      deleted: true, // Contract marked as deleted
+    } as Contract
+
+    const result = await getContractParams(
+      mockContract,
+      mockDb,
+      true, // Check access
+      'non-admin-user-id' // Non-admin user ID
+    )
+
+    expect(result).toEqual({ state: 'not found' })
+  })
+
+  // Test case for handling errors gracefully when external API calls fail
+  it('should handle errors gracefully when external API calls fail', async () => {
+    const mockDb = createClient('instanceId', 'key')
+    const mockContract = {
+      id: '123',
+      mechanism: 'cpmm-1',
+      outcomeType: 'BINARY',
+      visibility: 'private',
+      deleted: false,
+    } as Contract
+
+    jest
+      .spyOn(apiUtils, 'unauthedApi')
+      .mockImplementationOnce((path: string, params: any) => {
+        if (path === 'external-api-path') {
+          return Promise.reject(new Error('External API call failed'))
+        }
+        return Promise.resolve()
+      })
+
+    await expect(
+      getContractParams(mockContract, mockDb, true, undefined)
+    ).resolves.toMatchObject({
+      state: expect.any(String),
+    })
+  })
+
+  it('should correctly fetch and process CPMM contract parameters', async () => {
+    const mockDb = createClient('instanceId', 'key')
+    const mockContract = {
```

Comment:
Should there be a common place for mock contracts so they can be reused between tests? (And if we need to update it, because, for example, we added a new field, we would not have to update the mock contracts in a bunch of places.)